kani-driver/src/call_goto_synthesizer.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::util::warning;
+use anyhow::Result;
+use std::ffi::OsString;
+use std::path::Path;
+use std::process::Command;
+
+use crate::session::KaniSession;
+
+impl KaniSession {
+    /// Synthesize loop contracts for a goto binary `input` and produce a new goto binary `output`
+    /// The synthesizer we use is `goto-synthesizer` built in CBMC codebase, which is an enumerative
+    /// loop-contracts synthesizer. `goto-synthesizer` enumerates and checks if a candidate can be
+    /// used to prove some assertions, and return found invariants when all checks pass.
+    pub fn synthesize_loop_contracts(&self, input: &Path, output: &Path) -> Result<()> {
+        if !self.args.quiet {
+            warning("Start to synthesize loop contracts for all loops.");
+            warning("WARNING: this process may not terminate.");
+        }
+
+        let args: Vec<OsString> = vec![
+            "--loop-contracts-no-unwind".into(),
+            input.to_owned().into_os_string(),  // input
+            output.to_owned().into_os_string(), // output
+        ];
+
+        let mut cmd = Command::new("goto-synthesizer");
+        cmd.args(args);
+
+        self.run_suppress(cmd)?;
```

Comment:
Does the `goto-synthesizer` produce any output while it's running? If so, `run_terminal` might be more suitable.

Reply:
It only output debug messages when a high ```verbosity``` is specified.

Reply:
I see. In this case, I suggest doing:
```rust
       if self.args.quiet {
            self.run_suppress(cmd)?;
        } else {
            self.run_terminal(cmd)?;
        }
```
so that we can see its output if needed.

Reply:
Sounds great!